docs/guides/custom-functions.md

```diff
@@ -145,6 +145,68 @@ rules:
 
 Spectral would look for functions in a `my-functions` directory now.
 
+## Returning multiple results
+
+Spectral results are automatically deduplicated when they bear the same `code` and target the same `path`.
+As such, when your custom function is susceptible to return more than one result, you have to specify a different `path`
+for each result. 
+
+Below a sample function that checks tags bear unique names.
+
+**my-ruleset.yaml**
+
+```yaml
+functions: [uniqueTagNames]
+rules:
+  unique-tag-names:
+    message: "Tags should have distinct names: {{error}}"
+    given: "$.tags"
+    then:
+      function: "uniqueTagNames"
+```
+
+**functions/uniqueTagNames.js**
+
+```js
+const NAME_PROPERTY = 'name';
+
+module.exports = (targetVal, _opts, paths) => {
+
+    if (!Array.isArray(targetVal)) {
+        return;
+    }
+
+    const seen = [];
+    const results = [];
+
+    const rootPath = typeof (paths.target) !== 'undefined' ? paths.target : paths.given;
+
+    for (let i = 0; i < targetVal.length; i++) {
+        const tagName = targetVal[i][NAME_PROPERTY];
+
+        if (!tagName) {
+            continue;
+        }
+
+        if (seen.includes(tagName)) {
+            results.push(
+                {
+                    message: `Duplicate tag name '${tagName}'`,
+                    path: [...rootPath, i, NAME_PROPERTY]
+                },
+            );
+
+            console.error(results[results.length - 1]);
```

Comment:
```suggestion
```

Reply:
Let's not log anything.

Reply:
Duh... A forgotten witness of my crazy custom function debugging skills ;-)

Fixed